lib/bulk_insert/worker.rb

```diff
@@ -109,7 +109,10 @@ def compose_insert_query
           value = @now if value == :__timestamp_placeholder
 
           if ActiveRecord::VERSION::STRING >= "5.0.0"
-            value = @connection.type_cast_from_column(column, value) if column
+            if column
+              type = @connection.lookup_cast_type_from_column(column)
+              value = type.serialize(value)
```

Comment:
Can we confirm that this addition is backward compatible with Rails versions between 5 and 6.1?

Reply:
I run some tests locally and it seems to work.